### DISCUSSION TITLE
Pass arguments (e.g. current/selected commit or a branch hexsha) to external command?

### DISCUSSION THREAD

**Discussion by yarikoptic** [jesseduffield/lazygit#3840] [Q&A] [answered]
I am just trying out lazygit.  `gitk` is my primary helper for visualizing history etc. Great that I can run it upon `:`.  But if I could start it while also passing it e.g. selected branch while in branches view, or selected commit while in commits view (for `--select-commit` option)  -- that would have been  awesome.  I wonder if in general there is a way to pass such information into external commands?

**Comment by stefanhaller** [comment, marked as answer]
Interesting, this is the second time the suggestion comes up. See #3799 for more discussion.

Are you aware of custom commands (the ones you configure in the config file with a keybinding)? Might they help for what you want to do?

**Reply by yarikoptic** [reply]
no I wasn't!  I added one and it worked wonderfully (after restart of lazygit though):

```shell
❯ cat ~/.config/lazygit/config.yml
customCommands:
  - key: 'G'
    context: 'commits'
    command: 'gitk --select-commit="{{.SelectedLocalCommit.Hash}}"'
```
